### PR TITLE
0.5 release notes.

### PR DESCRIPTION
**Pull request by geeknoid**
These are the basic notes. Please:

- Make sure I didn't forget anything significant, or overstate trivial things.

- Look at the TODO items. Can you supply any missing doc links? If you own one of these features, considering adding a doc page to istio.io so we can link from it here.

Unfortunately, istio.io doesn't yet have updated adapter, template, and command-line reference documentation. Otherwise, I could point to these pages for several of the items. Hopefully, we'll have updated versions of these docs soon and I can come back and add links for these.

Thanks.

### CODE REVIEW

File: _about/notes/0.5.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+order: 96
+
+layout: about
+type: markdown
+---
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+update features detailed below.
```
Comment (vadimeisenbergibm) [inline _about/notes/0.5.md]: update -> updated

File: _about/notes/0.5.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Networking
+
+- **Alpha2 Traffic Rules**. Add support for an updated model of routing and destination rules. TODO: Doc link?
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
+
+- **Automatic Proxy Injection**. Use new Kubernetes mutating webhook functionality to provide automatic
```
Comment (vadimeisenbergibm) [inline _about/notes/0.5.md]: Available from Kubernetes 1.9.

File: _about/notes/0.5.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+updated features detailed below.
+
+## Networking
+
+- **Alpha2 Traffic Rules**. Add support for an updated model of routing and destination rules. TODO: Doc link?
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
+
+- **Automatic Proxy Injection**. Use new Kubernetes 1.9 mutating webhook functionality to provide automatic
```
Comment (linsun) [inline _about/notes/0.5.md]: We should note that auto side car injection will not work for k8s 1.7 and 1.8.

File: _about/notes/0.5.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+type: markdown
+---
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+updated features detailed below.
+
+## Networking
+
+- **Alpha2 Traffic Rules**. Add support for an updated model of routing and destination rules. TODO: Doc link?
```
Comment (linsun) [inline _about/notes/0.5.md]: is this ready to be announced?  @rshriram @ijsnellf @GregHanson

File: _about/notes/0.5.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+## Other
+
+- **Component Logging**. Istio components now offer a rich set of command-line options to control local logging, including
+common support for log rotation. 
+
+- **Consistent Version Reporting**. Istio components now offer a consistent command-line interface to report their version information.
+
+- **Optional Instance Fields**. Within configuration, definitions of Mixer instances no longer need to include every field of the
+associated template. Omitted fields get a zero or empty value.
+
```
Comment (linsun) [inline _about/notes/0.5.md]: Does it make sense to have a known issue section?  I think we should, especially for things that will work previously but it doesn't now.   I have got people complained to me when 0.4 goes out that we didn't warm people for ports change.

File: _about/notes/0.5.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+## Security
+
+- **Bring Your Own CA**. There have been many enhancements to the 'bring your own CA' feature. TODO: Doc link?
+
+- **PKCS8**. Add support for PKCS8 keys to Istio PKI.
+
+## Other
+
+- **Component Logging**. Istio components now offer a rich set of command-line options to control local logging, including
+common support for log rotation.
```
Comment (linsun) [inline _about/notes/0.5.md]: can we put a link to teach people how to view the logs?

File: _about/notes/0.5.md (hunk 5 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Sounds reasonable. Any recommendations for content?

File: _about/notes/0.5.md (hunk 6 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: We don't have docs for this yet. I'm working at the moment on extracting a fresh set of CLI reference docs to put on istio.io. Once we have those, we can point people to those docs.

**Comment by linsun** [comment]
@ldemailly +1, I was just thinking about that too. :)

File: _about/notes/0.5.md (hunk 6 above)
Comment (linsun) [reply _about/notes/0.5.md]: ok SGTM

File: _about/notes/0.5.md (hunk 5 above)
Comment (linsun) [reply _about/notes/0.5.md]: helm install is broken is one of them.  auto injection doesn't work on 1.7 or 1.8 is another one.

File: _about/notes/0.5.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+updated features detailed below.
+
+## Networking
+
+- **Alpha2 Traffic Rules**. Add support for an updated model of routing and destination rules. TODO: Doc link?
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
+
+- **Automatic Proxy Injection**. Use Kubernetes 1.9's mutating webhook functionality to provide automatic
+pod-level proxy injection. Automatic injection doesn't work requires Kubernetes 1.9 or beyond and
```
Comment (douglas-reid) [inline _about/notes/0.5.md]: punctuation missing between "Automatic injection doesn't work" and "requires Kubernetes 1.9" ?

File: _about/notes/0.5.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+common support for log rotation. 
+
+- **Consistent Version Reporting**. Istio components now offer a consistent command-line interface to report their version information.
+
+- **Optional Instance Fields**. Within configuration, definitions of Mixer instances no longer need to include every field of the
+associated template. Omitted fields get a zero or empty value.
+
+
+TODO: What should we say about multi-cluster support in this release?
+TODO: What should we say about Grafana/dashboards in this release?
```
Comment (douglas-reid) [inline _about/notes/0.5.md]: Does this release cut include both the Mixer and Pilot dashboards in the grafana image?  If so, I think we should mention that we have initial versions of dashboards for monitoring the Istio components themselves.

File: _about/notes/0.5.md (hunk 7 above)
Comment (ayj) [reply _about/notes/0.5.md]: Doc link: https://istio.io/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection

File: _about/notes/0.5.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+updated features detailed below.
+
+## Networking
+
+- **Alpha2 Traffic Rules**. Add support for an updated model of routing and destination rules. TODO: Doc link?
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
```
Comment (ayj) [inline _about/notes/0.5.md]: cc @ZackButcher

File: _about/notes/0.5.md (hunk 3 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done.

File: _about/notes/0.5.md (hunk 2 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done

File: _about/notes/0.5.md (hunk 1 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done

File: _about/notes/0.5.md (hunk 5 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done.

File: _about/notes/0.5.md (hunk 8 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: @ldemailly Who knows the answer to Doug's question?

File: _about/notes/0.5.md (hunk 7 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done

File: _about/notes/0.5.md (hunk 5 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Done

File: _about/notes/0.5.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+therefore doesn't work on older versions. [Learn more](https://istio.io/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection)
+
+## Mixer adapters
+
+- **Open Policy Agent**. Mixer now has an authorization adapter implementing the [open policy agent](http://www.openpolicyagent.org) model,
+providing a flexible fine-grained access control mechanism. TODO: Doc link?
+
+- **RBAC**. Mixer now has a resource-based access control adapter. TBD: Doc link?
+
+- **Fluentd**. Mixer now has an adapter for [fluentd](https://www.fluentd.org) telemetry collection. TODO: Doc link?
```
Comment (jeffmendoza) [inline _about/notes/0.5.md]: telemetry -> log (adapter only accepts logentries)
Doc link: https://istio.io/docs/tasks/telemetry/fluentd.html
Not posted yet. https://github.com/istio/istio.github.io/pull/905

File: _about/notes/0.5.md (hunk 4 above)
Comment (rshriram) [reply _about/notes/0.5.md]: Its partly ready. I would qualify that unlike alpha1, this is hot off the oven. Likely to be a bumpy ride

File: _about/notes/0.5.md (hunk 9 above)
Comment (rshriram) [reply _about/notes/0.5.md]: We need to highlight this and actually make some noise. This essentially handles the piecemeal installation.

File: _about/notes/0.5.md (hunk 4 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Should we remove it then? Or move it lower in the list? Different wording?

File: _about/notes/0.5.md (hunk 9 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Can someone supply better phrasing ?

File: _about/notes/0.5.md (hunk 4 above)
Comment (ZackButcher) [reply _about/notes/0.5.md]: Lower on the list, probably. We have support for ~70% of the new APIs, but it's likely we'll be spending quite a bit of time debugging/improving (they truly are "alpha" atm).

File: _about/notes/0.5.md (hunk 9 above)
Comment (ZackButcher) [reply _about/notes/0.5.md]: I'm not sure it's ready for prime time yet, the set of options it allows isn't as extensible as we'd like, and I don't think we have stable testing for it yet (e.g. we'd like to use it to deploy all of our e2e tests during presubmit, etc). cc @costinm thoughts here, you've been working in this space too.

File: _about/notes/0.5.md (hunk 4 above)
Comment (rshriram) [reply _about/notes/0.5.md]: Yeah. lower in the list.

File: _about/notes/0.5.md (hunk 9 above)
Comment (rshriram) [reply _about/notes/0.5.md]: Yeah, but its a step in the right direction. May be prefix it with something like a "preview" release or alpha version of Istio custom installer for folks seeking incremental adoption of Istio.

File: _about/notes/0.5.md (hunk 4 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: OK, I moved it down the list and reworded to add some caveats.

File: _about/notes/0.5.md (hunk 9 above)
Comment (ZackButcher) [reply _about/notes/0.5.md]: That's fair.

Comment (geeknoid) [reply _about/notes/0.5.md]: Zack, can you give me the phrasing to insert here?

Thanks.

File: _about/notes/0.5.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+type: markdown
+---
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+updated features detailed below.
+
+## Networking
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: @ZackButcher Any recommendations for this bullet?

File: _about/notes/0.5.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+therefore doesn't work on older versions. [Learn more]({{home}}/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection)
+
+- **Alpha2 Traffic Rules**. We've started to add support for a new model for routing and destination rules. You can see the updated
+definitions [here]({{home}}/docs/reference/config/istio.routing.v1alpha2.html). It's not quite ready to be used though, this is
+still very much a work in progress.
+
+## Mixer adapters
+
+- **Open Policy Agent**. Mixer now has an authorization adapter implementing the [open policy agent](http://www.openpolicyagent.org) model,
+providing a flexible fine-grained access control mechanism. TODO: Doc link?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: @wattli Do we have a doc link for this?

File: _about/notes/0.5.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+- **RBAC**. Mixer now has a resource-based access control adapter. TBD: Doc link?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: @wattli Do we have a doc link for this?

File: _about/notes/0.5.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+- **Fluentd**. Mixer now has an adapter for log collection through [fluentd](https://www.fluentd.org). 
+[Learn more]({{home}}/docs/tasks/telemetry/fluentd.html)
+
+- **Stdio**. The stdio adapter now lets you log to files with support for log rotation & backup, along with a host
+of controls.
+
+## Security
+
+- **Bring Your Own CA**. There have been many enhancements to the 'bring your own CA' feature. TODO: Doc link?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: Do we have a doc link for this or any more detail we can list directly in this doc?

File: _about/notes/0.5.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+associated template. Omitted fields get a zero or empty value.
+
+## Known issues
+
+- Installing with Helm chart is currently broken.
+
+- Automatic sidecar injection only works with Kubernetes 1.9 or later.
+
+
+TODO: What should we say about multi-cluster support in this release?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: @costinm @gnirodi Do we need to say anything in the release notes about multi-cluster?

File: _about/notes/0.5.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+TODO: What should we say about Grafana/dashboards in this release?
```
Comment (geeknoid) [inline _about/notes/0.5.md]: @hklai @douglas-reid It's not clear what Grafana functionality is actually include in the release. Andy, can you point us to the right thing so we can write appropriate release notes for this item?

File: _about/notes/0.5.md (hunk 14 above)
Comment (wattli) [reply _about/notes/0.5.md]: https://istio.io/docs/tasks/security/plugin-ca-cert.html

File: _about/notes/0.5.md (hunk 13 above)
Comment (wattli) [reply _about/notes/0.5.md]: https://docs.google.com/document/d/1dKXUEOxrj4TWZKrW7fx_A-nrOdVD4tYolpjgT8DYBTY/edit#

File: _about/notes/0.5.md (hunk 12 above)
Comment (wattli) [reply _about/notes/0.5.md]: https://docs.google.com/document/d/1U2XFmah7tYdmC5lWkk3D43VMAAQ0xkBatKmohf90ICA/edit#heading=h.fmlgl8m03gfy

File: _about/notes/0.5.md (hunk 16 above)
Comment (douglas-reid) [reply _about/notes/0.5.md]: from what I can tell, the release was cut at SHA: c9debce... . That includes the initial Mixer dash, but not Pilot (nor the fix to include the appropriate dashboard files via makefile).

so, my instinct is to not mention it at all in this release, and have it mentioned (with better docs in 0.6).

File: _about/notes/0.5.md (hunk 11 above)
Comment (rshriram) [reply _about/notes/0.5.md]: **Incremental Istio Deployment (Preview)**: You can now adopt Istio incrementally, more easier than before, by installing only the components you want (e.g, Pilot+Ingress only as the minimal Istio install). Refer to `istioctl` CLI tool for generating a customized Istio deployment.

File: _about/notes/0.5.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+title: Istio 0.5
+
+order: 96
+
+layout: about
+type: markdown
+---
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or
```
Comment (rshriram) [inline _about/notes/0.5.md]: May be a happy new year?

File: _about/notes/0.5.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+{% include home.html %}
+
+In addition to the usual pile of bug fixes and perf improvements, this release includes the new or 
+updated features detailed below.
+
+## Networking
+
+- **Control Plane Deployments**. `istioctl` now has a deployment command to produce istio control plane deployments. TODO: Doc link?
+
+- **Automatic Proxy Injection**. Use Kubernetes 1.9's mutating webhook functionality to provide automatic
```
Comment (rshriram) [inline _about/notes/0.5.md]: Webhook based proxy injection

File: _about/notes/0.5.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+pod-level proxy injection. Automatic injection requires Kubernetes 1.9 or beyond and
+therefore doesn't work on older versions. [Learn more]({{home}}/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection)
+
+- **Alpha2 Traffic Rules**. We've started to add support for a new model for routing and destination rules. You can see the updated
```
Comment (rshriram) [inline _about/notes/0.5.md]: Revised Traffic Rules: Based on user feedback, we have made significant changes to Istio's traffic management (routing rules, destination rules, etc.). We would love your early feedback while we polish this in the coming weeks....

File: _about/notes/0.5.md (hunk 18 above)
Comment (ayj) [reply _about/notes/0.5.md]: Do users care about the mechanism (webhook vs. initializer)? Manual vs. automatic is used elsewhere in the documentation. Alternatively, we could use client-side vs. server-side.

File: _about/notes/0.5.md (hunk 17 above)
Comment (geeknoid) [reply _about/notes/0.5.md]: Getting kind of late for that, don't you think?

**Comment by ldemailly** [comment]
it really helps subsequent review when commits are added instead of force pushing/losing the history at each pr update

File: _about/notes/0.5.md (hunk 5 above)
Comment (hklai) [reply _about/notes/0.5.md]: I believe mesh expansion is also broken. @costinm can provide more detail.

**Pull request merged**